### ISSUE TITLE
Error in VTK-based python example (PR #256)

### ISSUE THREAD

**Issue by julien-tierny** [topology-tool-kit/ttk#261]
**Describe the bug**
The latest VTK-based python example (PR #256) seems to run into issues with VTK 8.1

**To Reproduce**
1. Install TTK
2. Go to the following source directory example/python
3. Enter the following commands:
- export LD_LIBRARY_PATH="/usr/local/lib/"
- export PYTHONPATH="/usr/local/lib/python3.6/site-packages/"
- ./example.py ../data/inputData.vtu
4. The following output is reported:
Traceback (most recent call last):
  File "./example.py", line 30, in <module>
    curve.SetInputConnection(reader.GetOutputPort())
TypeError: SetInputConnection argument 1: method requires a VTK object

**Expected behavior**
The script should execute as the ParaView-based python example.

**Screenshots**
Not applicable.

**System (please complete the following information):**
 - OS: [e.g. iOS] gentoo linux 
 - Compiler [e.g. 22] gcc 8.2.0, python 3.6.9
 - Installed TTK's optional dependencies: all

**Additional context**
This one is for @lhofmann, who is the main author of PR #256.

**Comment by lhofmann** [comment]
Before I build VTK 8.1 (I tested it with VTK 8.2), could you check if adding `import vtk` at the top of example.py changes anything?

**Comment by julien-tierny** [comment]
unfortunately, it does not :( 
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Comment by lhofmann** [comment]
That's strange. TTK does not even build for me using VTK 8.1.2. Did you do an incremental build with a CMakeCache.txt from a previous version?

It works for me with the fix in #262

**Comment by julien-tierny** [comment]
still broken here.
I should have mentioned that before:
on my system, to build paraview plugins, I used the ttk patched paraview (5.6.1) installed in /usr/local. 
to build the vtk wrappers and standalone apps, I use a system-based installation of vtk 8.1.0 provided by my linux distribution (gentoo).
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Comment by lhofmann** [comment]
I'm not sure if this is even supported, since ParaView provides its own (non-compatible) copy of VTK.
In your python-path, there are probably `vtk.py` and `vtkmodules/`, which are coming from ParaView 5.6.0 (i.e., based on VTK 8.2), and also a `vtk/`, which is coming from VTK 8.1.
What is the output of `python -c 'import vtk; print(vtk.__file__)' `?

**Comment by julien-tierny** [comment]
> I'm not sure if this is even supported, since ParaView provides its own (non-compatible) copy of VTK.
> In your python-path, there are probably `vtk.py` and `vtkmodules/`, which are coming from ParaView 5.6.0 (i.e., based on VTK 8.2), and also a `vtk/`, which is coming from VTK 8.1.
I see. the "dual" support of vtk (system-based install + paraview-based install) is something that we extensively discussed back then with @CharlesGueunet

but this is the first time that we actually hit a concrete problem with that. commit 71907fc sets the advanced option TTK_FORCE_VTK_PARAVIEW to ON (to build against paraview-based vtk by default if present).
also, I had to revert PR #262 (was not working otherwise) and I added some debug info to have the console output matching pvpython's.

everything seems to be ok now.

cheers
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny
